### ISSUE TITLE
"-Xdump:none" generates core files

### ISSUE THREAD

**Issue by knn-k** [eclipse-openj9/openj9#15165]
When OpenJ9 is running with `-Xdump:none`, it generates a system dump file (core file on Linux) if the VM crashes.
It is explained in https://www.eclipse.org/openj9/docs/xdump/ as:

> If you remove all dump agents by using `-Xdump:none` with no further `-Xdump` options, the VM still provides these basic diagnostic outputs:
> - If a crash occurs, information about the location of the crash, VM options, and native and Java stack traces are written to stderr. **A system dump file is also written** to the user's home directory.

However, it is counterintuitive.
Specifying `-Xdump:system:none` turns off system dumps, then it is natural to expect `-Xdump:none` to turn off all the dumps (java, heap, jit, snap, system).

You can use the environment variable `J9NO_DUMPS` with `-Xdump:none` to stop system dumps.  But that is not documented, and it has been malfunctioning until fixed by PR #15122.

I would propose `-Xdump:none` to stop generating core files by default.

**Comment by keithc-ca** [comment]
I agree that it may be a little confusing that `-Xdump:none` and `-Xdump:system:none` have different behaviors, but that difference appears to be intentional.

Another difference is the behavior of `com.ibm.jvm.Dump.SystemDump()` which will produce a dump in the presence of `-Xdump:system:none`, but not if `-Xdump:none` is specified:
```
JVMJ9VM079W triggerOneOffDump(system) requires j9dmp29
```

Again, the differences appears to be intentional: closing.
